parser/src/main/java/io/specmesh/apiparser/model/Message.java

```diff
@@ -70,6 +77,50 @@ public class Message {
      * @return the location of the schema
      */
     public String schemaRef() {
-        return (String) payload.get("$ref");
+        final Object o = payload.get("$ref");
+        if (o == null) {
+            throw new APIParserException("$ref lookup not supported on this spec format");
+        }
+        return (String) o;
+    }
+
+    /**
+     * return properties: value: type: long value: type: $ref: "MessageValue.avsc". # avro-type (or
+     * json or proto)
+     *
+     * @return long or schema ref
+     */
+    public String valueType() {
+        return getType("value");
+    }
+
+    /**
+     * message: payload: properties: key: type: string # in build kafka-type - one of UUID, long,
+     * int, short, float, double, string, bytes value: type: double
+     *
+     * @return 'string'
+     */
+    public String keyType() {
+        return getType("key");
+    }
+
+    private String getType(final String lookup) {
+        final var typeMaybe = ((Map) ((Map) payload.get("properties")).get(lookup)).get("type");
```

Comment:
This will throw NPE if any of these properties are missing.

Reply:
true - thought Id wrapped it